### PR TITLE
Test for checking null contents_changed after repo sync

### PR DESCRIPTION
**Pull request by ColeHiggins2**
New customer scenario for testing contents_changed after repo sync

### CODE REVIEW

**Comment by ColeHiggins2** [comment]
Confirm functionality with dev

**Comment by ColeHiggins2** [comment]
When running the contents_changed command in the CLI, I get 0 rows returned( This is what we want)
```
could not change directory to "/root": Permission denied
 class | execution_plan_uuid | input 
-------+---------------------+-------
(0 rows)
```
However, when running the command via target_sat.execute() I get an empty string. Am I correct in assuming this is the same as (0 rows)?

File: tests/foreman/api/test_repository.py [hunk 1]
```diff
@@ -1425,6 +1425,42 @@ def test_positive_bulk_cancel_sync(self, target_sat, module_manifest_org):
...
+            upload_manifest(module_org.id, manifest.content)
+        repo_id = enable_rhrepo_and_fetchid(
+            basearch='x86_64',
+            org_id=module_org.id,
+            product=constants.PRDS['rhel'],
+            repo=constants.REPOS['rhst7']['name'],
+            reposet=constants.REPOSET['rhst7'],
+            releasever=None,
+        )
+        entities.Repository(id=repo_id).sync()
```
Comment (mshriver) [inline tests/foreman/api/test_repository.py:1455]: Please use `target_sat.api.Repository()`

File: tests/foreman/api/test_repository.py [hunk 2]
```diff
@@ -1425,6 +1425,42 @@ def test_positive_bulk_cancel_sync(self, target_sat, module_manifest_org):
...
+        :CaseImportance: High
+
+        :customerscenario: true
+
+        :BZ: 2089580
+
+        :CaseAutomation: Automated
+        """
+        with manifests.clone() as manifest:
+            upload_manifest(module_org.id, manifest.content)
```
Comment (mshriver) [inline tests/foreman/api/test_repository.py:1446]: https://github.com/SatelliteQE/robottelo/blob/0018520356b29fe680bb931791bc55f17f0a76f0/pytest_fixtures/component/taxonomy.py#L62

Looks like this fixture would be appropriate for the test so you don't have to handle the manifest within the call phase.

**Comment by mshriver** [comment]
@ColeHiggins2 concerning your comment about the return from the `psql` command execution, sounds fine to me. Its a matter of the defaults for the table (or lack of) output when running the command directly, versus the formatting that hammer CLI is doing.

You can add an additional check against the return code of the command - to make sure that it was successful and returned nothing to stdout.

**Review by latran — CHANGES_REQUESTED** [review]
Pending test run result after changes.

File: tests/foreman/api/test_repository.py [hunk 3]
```diff
@@ -1425,6 +1425,42 @@ def test_positive_bulk_cancel_sync(self, target_sat, module_manifest_org):
...
+        repo_id = enable_rhrepo_and_fetchid(
+            basearch='x86_64',
+            org_id=module_org.id,
+            product=constants.PRDS['rhel'],
+            repo=constants.REPOS['rhst7']['name'],
+            reposet=constants.REPOSET['rhst7'],
+            releasever=None,
+        )
+        entities.Repository(id=repo_id).sync()
+        assert 1 == 1
```
Comment (Griffin-Sullivan) [inline tests/foreman/api/test_repository.py:1456]: What's the purpose of this assertion?

Comment (ColeHiggins2) [reply tests/foreman/api/test_repository.py]: accidentally left in there. oops

**Comment by latran** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_repository.py -k test_positive_sync_repo_null_contents_changed

File: tests/foreman/api/test_repository.py [hunk 4]
```diff
@@ -1425,6 +1425,39 @@ def test_positive_bulk_cancel_sync(self, target_sat, module_manifest_org):
...
+            reposet=constants.REPOSET['rhst7'],
+            releasever=None,
+        )
+        target_sat.api.Repository(id=repo_id).sync()
+        prod_log_out = target_sat.execute(
+            "sudo -u postgres psql -d foreman -c 'select class,execution_plan_uuid,input "
+            "from dynflow_actions where input LIKE '%\'contents_changed\':null%' AND class = "
+            "'Actions::Katello::Repository::Sync';'"
+        )
+        assert prod_log_out.stdout == ''
```
Comment (latran) [inline tests/foreman/api/test_repository.py:1459]: Can you also add a `prod_log_out.status`?  It might be a better output than `stdout`.

**Review by latran — CHANGES_REQUESTED** [review]

File: tests/foreman/api/test_repository.py [hunk 5]
```diff
@@ -1451,6 +1451,41 @@ def test_positive_sync_sha_repo(self, repo, target_sat):
         )
         assert result.status == 1
 
+    @pytest.mark.tier2
+
```
Comment (mshriver) [inline tests/foreman/api/test_repository.py:1455]:
```suggestion

```

**Review by mshriver — APPROVED** [review]

File: tests/foreman/api/test_repository.py (hunk 5 above)
Comment (mshriver) [reply tests/foreman/api/test_repository.py:1455]: This will fix pre-commit failures.

**Review by latran — APPROVED** [review]

**Review by Griffin-Sullivan — APPROVED** [review]

**Pull request merged**